components/central-application-gateway/pkg/httptools/logging.go

```diff
@@ -1,28 +1,75 @@
 package httptools
 
 import (
+	"context"
 	"net/http"
 	"time"
 
-	log "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/uuid"
 )
 
+type ContextKey string
+
+const ContextUUID ContextKey = "context-uuid"
+
+func LogResponse(log *zap.SugaredLogger, res *http.Response) error {
+	req := res.Request
+	log.WithOptions(
+		zap.AddCallerSkip(1),
+	).
+		With(
+			"requestID", req.Context().Value(ContextUUID),
+			"method", req.Method,
+			"host", req.Host,
+			"url", req.URL.RequestURI(),
+			"proto", req.Proto,
+			"code", res.StatusCode,
+			"contentLength", res.ContentLength,
+			"kind", "response",
+		).Debugf("%s %s %s %s", req.Method, req.Host, req.URL.RequestURI(), req.Proto)
+
+	return nil
+}
+
+func LogRequest(log *zap.SugaredLogger, r *http.Request) {
+	log.WithOptions(
+		zap.AddCallerSkip(1),
+	).With(
+		"requestID", r.Context().Value(ContextUUID),
+		"method", r.Method,
+		"host", r.Host,
+		"url", r.URL.RequestURI(),
+		"proto", r.Proto,
+		"kind", "request",
+	).Debugf("%s %s %s %s", r.Method, r.Host, r.URL.RequestURI(), r.Proto)
+}
+
 func RequestLogger(label string, h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		uid := uuid.NewUUID()
+
+		r = r.WithContext(context.WithValue(r.Context(), ContextUUID, uid))
+
 		lw := newLoggingResponseWriter(w)
 
 		h.ServeHTTP(lw, r)
 
-		method := r.Method
 		fullPath := r.RequestURI
 		if fullPath == "" {
 			fullPath = r.URL.RequestURI()
 		}
-		proto := r.Proto
 		responseCode := lw.status
 		duration := time.Since(lw.start).Nanoseconds() / int64(time.Millisecond)
 
-		log.Infof("%s: %s %s %s %d %d", label, method, fullPath, proto, responseCode, duration)
+		log := zap.L().Sugar().
```

Comment:
Suggestion: can we also log the`fullPath` variable in default logger? Or if you see a reason not to log this variable by default could you delete the `fullPath` variable as it is not used

Reply:
I am not sure what you mean? `LogRequest` already logs `req.URL.RequestURI()`

Reply:
The `fullPath` variable is not used in this function

Reply:
isn't `fullPath` equivalent to `r.URL.RequestURI()`?